gtars/src/uniwig/mod.rs

```diff
@@ -1,16 +1,722 @@
+
+
+                                },
+                                _ => {println!("Default to wig file.")},
+                            }
+
+                    },
+                    _ => println!("Unexpected value: {}", j), // Handle unexpected values
+                }
+            }
+        }
+    }
+
+
+
+
+
+
+
+}
+
+fn write_to_npy_file(counts: &Vec<u32>, filename: String, chromname: String, start_position: i32, stepsize: i32, metafilename: String) {
+
+    // For future reference `&Vec<u32>` is a SLICE and thus we must use the `to_vec` function below when creating an array
+    // https://users.rust-lang.org/t/why-does-std-to-vec-exist/45893/9
+
+    println!("{}", filename);
+    println!("{}", metafilename);
+
+    // Write the NumPy Files
+    let arr = Array::from_vec(counts.to_vec());
+    let _ = write_npy(filename, &arr);
+
+    // Write to the metadata file.
+    // Note: there should be a single metadata file for starts, ends and core
+
+    let mut file = OpenOptions::new()
+        .create(true)  // Create the file if it doesn't exist
+        .append(true)  // Append data to the existing file if it does exist
+        .open(metafilename).unwrap();
+
+    // The original wiggle file header. This can be anything we wish it to be. Currently space delimited.
+    let wig_header = "fixedStep chrom=".to_string() + chromname.as_str() + " start="+start_position.to_string().as_str() +" step="+stepsize.to_string().as_str();
+    file.write_all(wig_header.as_ref()).unwrap();
+    file.write_all(b"\n").unwrap();
+
+}
+
+#[allow(unused_variables)]
+fn write_to_wig_file(counts: &Vec<u32>, filename: String, chromname: String, start_position: i32, stepsize: i32) {
+
+    let mut file = OpenOptions::new()
+        .create(true)  // Create the file if it doesn't exist
+        .append(true)  // Append data to the existing file if it does exist
+        .open(filename).unwrap();
+
+    //println!("DEBUG: fixedStep chrom={}",chromname.clone());
+    let wig_header = "fixedStep chrom=".to_string() + chromname.as_str() + " start="+start_position.to_string().as_str() +" step="+stepsize.to_string().as_str();
+    file.write_all(wig_header.as_ref()).unwrap();
+    file.write_all(b"\n").unwrap();
+
+    let mut position = 0;
+
+    for count in counts.iter(){
+        //TODO THis is inefficient to iterate over ALL counts when the above coordinate vecs could act as an index
+        if *count == 0 {
+            position += 1;
+            continue
+        } else{
+
+            //println!("DEBUG COORDINATE = {} COUNTS= {}",position, count);
+            //let wig_line = position.to_string() + " " + count.to_string().as_str();
+            let wig_line = count.to_string();
+            file.write_all(wig_line.as_ref()).unwrap();
+            file.write_all(b"\n").unwrap();
+            position+=1;
+        }
+
+    }
+
+
+}
+
+fn read_chromosome_sizes(chrom_size_path: &str) -> Result<std::collections::HashMap<String, u32>, Box<dyn Error>> {
```

Comment:
I think I have `anyhow` as a dependency in this crate, so you can actually use `anyhow::Result` here instead which is effectively the same thing as `Result<T, Box<dyn Errror>>`